week2/Homework/Json/j_index.js

```diff
@@ -0,0 +1,109 @@
+//console.log("Hello World!" , process.argv);
+fs = require('fs');
+let command = "";
+
+if (process.argv.length < 3)
+	command = "help" ;
+else
+	command = process.argv[2];
+
+function readFile(file_name){return new Promise(
+    (resolve, reject) => {
+		fs.readFile(file_name, 'utf8', function (err,data) {
+			  if (err) {
+				reject(err);
+			  }
+			  else{
+				  resolve(data);
+			  }
+		});
+    });
+}
+function writeFile(file_name , file_data){return new Promise(
+    (resolve, reject) => {
+		fs.writeFile(file_name, file_data, function (err) {
+			  if (err) {
+				reject(err);
+			  }
+			  else{
+				resolve(file_data);
+			  }
+		});
+    });
+}
+
+switch(command){
+	case "help" :
+		readFile('j_help.txt')
+		.then((result) => console.log(result))
+		.catch((error) => console.log(error));
+		break;
+	case "list":
+		readFile('j_items.txt')
+		.then((result) => console.log(result))
+		.catch((error) => console.log(error));
+		break;
+	case "add":
+		if (process.argv.length >= 5){

```

Comment:
Comments would be helpful here. I see that you're using process.argv.length, but without a lot of effort reading your code, it's not super clear why. What happens if process.argv.length is less than 5? (from your code, it looks like nothing will happen - no error message, just nothing).

I'd like to see something like:
```
case "add":
    // usage: node index.js add 3 'this is the text of my stuff'
    ...code...
```